src/ripple/basics/impl/make_SSLContext.cpp

```diff
@@ -154,16 +163,28 @@ static void x509_sign (X509* x509, EVP_PKEY* evp_pkey)
         LogicError ("X509_sign failed");
 }
 
-static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr& cert)
+static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr cert)
 {
-    if (SSL_CTX_use_certificate (ctx, cert.release()) <= 0)
+    if (SSL_CTX_use_certificate (ctx, cert.get()) <= 0)
         LogicError ("SSL_CTX_use_certificate failed");
+    {
+        static std::mutex m;
+        static x509_ptr cleanupMem;
+        std::lock_guard<std::mutex> l{m};
```

Comment:
@HowardHinnant The reason I can't do that is these functions are called multiple times in the unit tests. With that change it would leak all but the first one. As written it destroys the first one and keeps the new one unit the next init (or shutdown).

I think I'm going to replace the `static` with a `thread_local` and get rid of the `mutex` - that's an easy change.